src/context/mod.rs

```diff
@@ -31,43 +33,31 @@ pub use provided_types::{
 pub use skip_options::{SkipOptions, SkipOptionsBuilder};
 
 use crate::{
+    environment::Environment,
+    fixing::PendingFix,
     rule::InstantiatedRule,
     text::get_text_slice,
     tree_sitter::{Language, Node, Query},
     violation::{Violation, ViolationWithContext},
-    AggregatedQueries, Config, SourceTextProvider,
+    AggregatedQueries, Config, Fixer, FixingForSliceRunContext, SourceTextProvider,
 };
 
+#[derive(Copy, Clone)]
 pub struct FileRunContext<'a, 'b> {
     pub path: &'a Path,
     pub file_contents: RopeOrSlice<'a>,
     pub tree: &'a Tree,
     pub config: &'a Config,
-    pub language: SupportedLanguage,
+    pub supported_language_language: SupportedLanguageLanguage,
     pub(crate) aggregated_queries: &'a AggregatedQueries<'a>,
     pub(crate) query: &'a Arc<Query>,
+    #[allow(dead_code)]
     pub(crate) instantiated_rules: &'a [InstantiatedRule],
+    #[allow(dead_code)]
     changed_ranges: Option<&'a [Range]>,
     from_file_run_context_instance_provider: &'b dyn FromFileRunContextInstanceProvider<'a>,
-}
-
-impl<'a, 'b> Copy for FileRunContext<'a, 'b> {}
-
-impl<'a, 'b> Clone for FileRunContext<'a, 'b> {
-    fn clone(&self) -> Self {
-        Self {
-            path: self.path,
-            file_contents: self.file_contents,
-            tree: self.tree,
-            config: self.config,
-            language: self.language,
-            aggregated_queries: self.aggregated_queries,
-            query: self.query,
-            instantiated_rules: self.instantiated_rules,
-            changed_ranges: self.changed_ranges,
-            from_file_run_context_instance_provider: self.from_file_run_context_instance_provider,
-        }
-    }
+    pub run_kind: RunKind<'a>,
```

Comment:
Ok right I started exposing this so that lint rules could have visibility of it, don't remember exactly where I've tried using it so far